Build: Enable PR testing with browserstack-runner

jzaefferer opened the pull request:
This runs our unit tests on all supported browsers for all commits, including pull requests, using the existing `browserstack-runner` configuration. A [sample run](https://travis-ci.org/jzaefferer/qunit/builds/45673762) took 2 1/2 minutes to test on 14 browsers.

This works, but I expect that we'll get the occasional failure, likely with one or more workers timing out, [like this](https://travis-ci.org/jzaefferer/qunit/builds/45670237). I think thats an acceptable tradeoff. We'll continue working with BrowserStack to make testing more reliable.

As for the obscured credentials: Those are easy enough to decode, the point is to avoid them being searchable, that is a search for "browserstack" on GitHub won't immediately show these credentials. This is a temporary solution until Travis CI [implements a proper one](https://github.com/travis-ci/travis-ci/issues/1946).

Once this lands I'm going to apply this for Globalize as well ([configuration is already in place](https://github.com/jquery/globalize/pull/367)). From there we can see how to move on, to eventually shut down TestSwarm and maybe replace Sizzle's use of Karma.

jzaefferer commented:
As if to prove my concern, the Travis build for this PR failed, due to one worker timing out.

jzaefferer commented:
I'm in touch with BrowserStack folks and they said those failures are issues on their end, which they're looking into. I hope we can get those ironed out and get consistent stable builds, where unlike TestSwarm, reruns won't be necessary.

leobalter commented:
> As for the obscured credentials: Those are easy enough to decode, the point is to avoid them being searchable, that is a search for "browserstack" on GitHub won't immediately show these credentials. This is a temporary solution until Travis CI implements a proper one.

This could be placed on the commit's description in case we land it this way, or at least a short description.

Looking forward to see it able to ship.

jzaefferer commented:
I've updated the commit message. I've removed or replaced the word "browserstack" for obvious reasons.

Haven't yet heard more from them, will ping them on Friday if nothing happens till then.

leobalter commented:
Is there any update from them?

jzaefferer commented:
There are still some issues, but those turned out very difficult to reproduce. I did a Skype call with one of them, triggering builds while he was debugging it live, but we still didn't get anywhere. Once he's back from his vacation, we'll try again.

jzaefferer commented:
Did some more test runs, and everything was working fine, both locally and on Travis. Let's see how it works in practice.
